api/src/api/resource.js

```diff
@@ -88,18 +97,48 @@ router.get(
         orderBy = { name: 1 };
       }
     }
-    const aggregation =
-      orderBy === null
-        ? [addFields, { $match: query }]
-        : [addFields, { $match: query }, { $sort: orderBy }];
-
+    function getAggregation() {
+      if (orderBy === null) {
+        return [
+          {
+            $facet: {
+              totalData: [addFields, { $match: query }],
+            },
+          },
+        ];
+      } else if (page === undefined || size === undefined) {
```

Comment:
Great! I think just to have more of a "catch all", we can do `page == null || size == null`, which encompasses both null or undefined